Example/Tests/FullStoryMiddlewareTest.m

```diff
@@ -0,0 +1,182 @@
+                                                             context:emptyDict
+                                                        integrations:emptyDict];
+    SEGTrackPayload* expect = [[SEGTrackPayload alloc] initWithEvent:@"testEvent"
+                                                          properties:@{@"FSSessionURL": @"testURL"}
+                                                             context:emptyDict
+                                                        integrations:emptyDict];
+    SEGContext *context = [[[SEGContext alloc] initWithAnalytics:[SEGAnalytics sharedAnalytics]] modify:^(id<SEGMutableContext> _Nonnull ctx) {
+        ctx.eventType = SEGEventTypeTrack;
+        ctx.payload = trackPayload;
+    }];
+    
+    SEGTrackPayload* output = (SEGTrackPayload*)[_fullStoryMiddleware getNewPayloadWithFSURL:context];
+
+    XCTAssertEqualObjects(expect.context, output.context);
+    XCTAssertEqualObjects(expect.properties, output.properties);
+    XCTAssertEqualObjects(expect.integrations, output.integrations);
+    XCTAssertEqualObjects(expect.event, output.event);
+    XCTAssertEqualObjects(expect.timestamp, output.timestamp);
+}
+
+- (void)testFullStoryMiddleware_GetNewPayloadWithFSURL_ScreenPayload_ReturnsScreenPayloadType {
+    // create a mock for the SEGContext, we only need eventType from it here
+    id segContextMock = OCMClassMock([SEGContext class]);
+    OCMStub([segContextMock eventType]).andReturn(SEGEventTypeScreen);
+   
+    SEGPayload* output = [_fullStoryMiddleware getNewPayloadWithFSURL:segContextMock];
+    XCTAssertTrue([output isKindOfClass:[SEGScreenPayload class]]);
+}
+
+- (void)testFullStoryMiddleware_GetNewPayloadWithFSURL_ScreenPayload_ReturnsScreenPayloadWithFSURL {
+    // create a mock for the FS
+    id fs = OCMClassMock([FS class]);
+    OCMStub([fs currentSessionURL]).andReturn(@"testURL");
+
+    NSDictionary* emptyDict = [[NSDictionary alloc] init];
+    SEGScreenPayload* screenPayload = [[SEGScreenPayload alloc] initWithName:@"testScreen"
+                                                          properties:emptyDict
+                                                             context:emptyDict
+                                                        integrations:emptyDict];
+    SEGScreenPayload* expect = [[SEGScreenPayload alloc] initWithName:@"testScreen"
+                                                          properties:@{@"FSSessionURL": @"testURL"}
+                                                             context:emptyDict
+                                                        integrations:emptyDict];
+    SEGContext *context = [[[SEGContext alloc] initWithAnalytics:[SEGAnalytics sharedAnalytics]] modify:^(id<SEGMutableContext> _Nonnull ctx) {
+        ctx.eventType = SEGEventTypeScreen;
+        ctx.payload = screenPayload;
+    }];
+    
+    SEGScreenPayload* output = (SEGScreenPayload*)[_fullStoryMiddleware getNewPayloadWithFSURL:context];
```

Comment:
I see you doing (what seems to be the same lines of code I could be wrong) in
`testFullStoryMiddleware_GetNewPayloadWithFSURL_TrackPayload_ReturnsTrackPayloadWithFSURL ` 
&
`testFullStoryMiddleware_GetNewPayloadWithFSURL_ScreenPayload_ReturnsScreenPayloadWithFSURL` 
on lines: `76-90` & `113-125`

Is it possible to abstract this out into a method below that returns a tuple of `expect` and `context` for your XCAssert tests below?

Reply:
Great point! They are very similar (just different payload type). objC unfortunately doesn't have tuples, but group them  into some kind of util method(s) would a lot of sense!

Reply:
Oh thats right >.< im so spoiled with swift.